### PR TITLE
fix: codeql fixes

### PR DESCRIPTION
**Pull request by hallvictoria**
## Description

Fixes #
Made changes in the tests/ folder to address CodeQL errors.

1. Weak hashes in blob function apps: replaced md5 references with sha256
2. [Reflected server-side cross-site scripting](https://codeql.github.com/codeql-query-help/python/py-reflective-xss/): added sanitization on HTTP request body
3. [URL redirection from remote source](https://codeql.github.com/codeql-query-help/python/py-url-redirection/): added validation on URL
4. [Usage of unapproved crypto library](https://codeql.github.com/codeql-query-help/python/py-weak-cryptographic-algorithm/): updated pyproject.toml to use pycryptodome instead of pycrypto
---

### PR information

- [ ] The title of the PR is clear and informative.
- [ ] There are a small number of commits, each of which has an informative message. This means that previously merged commits do not appear in the history of the PR. For information on cleaning up the commits in your pull request, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).
- [ ] If applicable, the PR references the bug/issue that it fixes in the description.
- [ ] New Unit tests were added for the changes made and CI is passing.

### Quality of Code and Contribution Guidelines
- [ ] I have read the [contribution guidelines](https://github.com/Azure/azure-functions-python-worker/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: tests/unittests/test_third_party_http_functions.py [hunk 1]
```diff
@@ -5,6 +5,8 @@
 import pathlib
 import re
 import typing
+import urllib.parse
```
Comment (gavin-aguiar) [inline tests/unittests/test_third_party_http_functions.py:8]: Why this change?

**Review by gavin-aguiar — APPROVED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -64,6 +64,7 @@ dev = [
     "ptvsd",
     "python-dotenv",
     "plotly",
+    "pycryptodome",
```
Comment (gavin-aguiar) [inline pyproject.toml]: This is already in the list

Comment (hallvictoria) [reply pyproject.toml]: Removed
